service/tests/build_crate.rs

```diff
@@ -31,13 +32,30 @@ fn not_shuttle() {
 }
 
 #[test]
-#[should_panic(
-    expected = "a cdylib was not created. Try adding the following to the Cargo.toml of the service:\n[lib]\ncrate-type = [\"cdylib\"]\n"
-)]
+fn not_lib() {
+    let buf = Box::new(DummyWriter {});
+    let project_path = format!("{}/tests/resources/not-lib", env!("CARGO_MANIFEST_DIR"));
+    assert!(build_crate(Path::new(&project_path), buf).is_err());
```

Comment:
Maybe we should `unwrap()` + `should_panic` to ensure the error is the correct one?